### PR TITLE
[Feature](aggregation) add agg_state define and ddl support

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

add agg_state define and ddl support.
```sql
  create table a_table(
      k1 int null,
      k2 agg_state max_by(int not null,int)
  )
  aggregate key (k1)
  distributed BY hash(k1) buckets 3
  properties("replication_num" = "1","disable_auto_compaction" = "true");

```

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run feut

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/catalog/PrimitiveType.java [hunk 1]
```diff
@@ -1141,7 +1151,7 @@ public boolean isVariantType() {
     }
 
     public boolean isStringType() {
-        return (this == VARCHAR || this == CHAR || this == HLL || this == STRING);
```
Comment (HappenLee) [inline fe/fe-common/src/main/java/org/apache/doris/catalog/PrimitiveType.java:1144]: why remove hll？

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DescribeStmt.java [hunk 2]
```diff
@@ -199,7 +200,11 @@ public void analyze(Analyzer analyzer) throws UserException {
                             // Extra string (aggregation and bloom filter)
                             List<String> extras = Lists.newArrayList();
                             if (column.getAggregationType() != null) {
-                                extras.add(column.getAggregationType().name());
+                                if (column.getAggregationType() == AggregateType.GENERIC_AGGREGATION) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DescribeStmt.java:203]: a new virtual getAggregationString do：
1. type check
2. get right string

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/IndexSchemaProcNode.java [hunk 3]
```diff
@@ -59,7 +60,11 @@ public ProcResult fetchResult() throws AnalysisException {
             // Extra string (aggregation and bloom filter)
             List<String> extras = Lists.newArrayList();
             if (column.getAggregationType() != null) {
-                extras.add(column.getAggregationType().name());
+                if (column.getAggregationType() == AggregateType.GENERIC_AGGREGATION) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/IndexSchemaProcNode.java:63]: same to up

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**